src/commands/revisionCommands/changeRevisionActiveState.ts

```diff
@@ -4,18 +4,22 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { ContainerAppsAPIClient } from '@azure/arm-app';
-import { IActionContext } from "@microsoft/vscode-azext-utils";
+import { IActionContext, nonNullProp } from "@microsoft/vscode-azext-utils";
 import { ext } from "../../extensionVariables";
+import { ContainerAppTreeItem } from '../../tree/ContainerAppTreeItem';
 import { RevisionTreeItem } from "../../tree/RevisionTreeItem";
 import { createContainerAppsAPIClient } from "../../utils/azureClients";
 import { localize } from "../../utils/localize";
-import { nonNullValue } from "../../utils/nonNull";
 
-export async function changeRevisionActiveState(context: IActionContext, command: 'activate' | 'deactivate' | 'restart', node?: RevisionTreeItem): Promise<void> {
+export async function changeRevisionActiveState(context: IActionContext, command: 'activate' | 'deactivate' | 'restart', node?: ContainerAppTreeItem | RevisionTreeItem): Promise<void> {
     if (!node) {
-        node = await ext.tree.showTreeItemPicker<RevisionTreeItem>(RevisionTreeItem.contextValue, context);
+        node = await ext.tree.showTreeItemPicker<ContainerAppTreeItem | RevisionTreeItem>(ContainerAppTreeItem.contextValue, context);
     }
 
+    const containerAppName: string = node instanceof RevisionTreeItem ? node.parent.parent.name : node.name;
+    const revisionName: string = node instanceof RevisionTreeItem ? node.name : nonNullProp(node.data, 'latestRevisionName');
+    const resourceGroupName: string = node instanceof RevisionTreeItem ? node.parent.parent.resourceGroupName : node.resourceGroupName;
```

Comment:
I think it would be cleaner to do something like this:

```ts
let containerAppName: string;
...
if (node instanceof RevisionTreeItem) {
    containerAppName = ...
    ...
} else {
    ...
}
```

But I know you're trying to release and this isn't a blocker.

Reply:
I actually went back and forth on it.  That way is more clear what is happening, but it takes like 3x more lines of code XD